### PR TITLE
icu: update to 65.1

### PR DESCRIPTION
**Pull request by nxhack**
Maintainer: me
 Compile tested: ath79, aarch64, trunk r11167-273a6cb
 Run tested: aarch64 (qemu)

Description:
ICU 65 updates to CLDR 36 locale data with many additions and corrections, and some new measurement units. The Java LocaleMatcher API is improved, and ported to C++. For building ICU data, there are new filtering options, and new tracing support for data loading in ICU4C.

Signed-off-by: Hirokazu MORIKAWA <morikw2@gmail.com>

### CODE REVIEW

**Comment by neheb** [comment]
I would remove the uClibc-ng patch. It's totally wrong and gives wrong results (std:: variant has multiple return values).

**Comment by nxhack** [comment]
OK, removed.

File: libs/icu/Makefile [hunk 1]
```diff
@@ -8,14 +8,14 @@
...
 PKG_VERSION:=$(MAJOR_VERSION).$(MINOR_VERSION)
-PKG_RELEASE:=2
+PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(MAJOR_VERSION)_$(MINOR_VERSION)-src.tgz
 PKG_SOURCE_URL:=https://github.com/unicode-org/icu/releases/download/release-$(MAJOR_VERSION)-$(MINOR_VERSION)
-PKG_HASH:=627d5d8478e6d96fc8c90fed4851239079a561a6a8b9e48b0892f24e82d31d6c
+PKG_HASH:=53e37466b3d6d6d01ead029e3567d873a43a5d1c668ed2278e253b683136d948
 
 PKG_LICENSE:=ICU-1.8.1+
```
Comment (neheb) [inline libs/icu/Makefile]: I'd replace this with just ICU. https://spdx.org/licenses/ has ICU as the tag.

**Comment by neheb** [comment]
PKG_BUILD_DIR seems unnecessary as well.

**Comment by yousong** [comment]
> I would remove the uClibc-ng patch. It's totally wrong and gives wrong results (std:: variant has multiple return values).

Hi @neheb , what does it mean "std:: variant has multiple return values"?  Can you elaborate?

By the way, QEMU has a similar patch, https://github.com/openwrt/packages/blob/master/utils/qemu/patches/0003-disas-fix-compilation-failure-when-isnan-is-a-macro.patch .  Does it also have the same issue?

**Comment by neheb** [comment]
@yousong https://en.cppreference.com/w/cpp/numeric/math/round

Note that std::round returns either float, double, or long double depending on the argument passed.

round in C returns double.

**Comment by nxhack** [comment]
> That patch is correct. Disregard what I said.
revert :)

**Comment by neheb** [comment]
@nxhack I was talking about the QEMU patch in that entire comment.

The uClibc-ng patch is still wrong. I have no motivation to try and come up with a proper solution.

**Comment by nxhack** [comment]
revert^2

**Comment by neheb** [comment]
Merged.

**Pull request merged**